daffodil-core/src/main/scala/org/apache/daffodil/core/dsom/RestrictionUnion.scala

```diff
@@ -102,10 +107,10 @@ final class Restriction private (xmlArg: Node, val simpleTypeDef: SimpleTypeDefB
    * Exclusive - restriction either has a baseType or a direct primType.
    */
   lazy val (optDirectPrimType, optBaseTypeDef: Option[GlobalSimpleTypeDef]) = {
-    val optPT = schemaSet.getPrimitiveType(baseQName)
+    val optPT = PrimType.fromQName(baseQName)
```

Comment:
> The schemaSet.getPrimitiveType(baseQName) should take the baseQName and run up it's definition chain until it finds the eventual primitive type it is derived from

That's not what getPrimitiveType currently does. It has basically the same logic as the new `PrimType.fromQName` in that it returns a `Some` if the QName is the XSD namspace and the localName is one of the primitive types, and it returns None otherwise.  We don't currently have function that does the chain thing to find a primitive type. Right now you would have call getGlobaSimpleTypeDef and then look at primType on the result to find the base primitive type. And if a property can resolve to a global simple type def OR a primitive type, you can to call both getGlobalSImpleTypeDef and getPrimType.

We could change add a new function that could do what you describe, but it's maybe outside the scope of this change?

> Is this code really trying to both chase the chain back to the primitive base type while also assuring that the referenced types all exist?

It looks like restriction unions have different logic depending on if the type is a primitive type or a global simple type def, looks like something about combining facets from different type defs? I'm not too sure, as I didn't really look to deep into this since I didn't change the logic, just changed how resolve a string to a QName. What we do with that QName hasn't changed.

Note that I forget exactly why I moved fromQName from SchemaSet to PrimType (I went down a few different rabbit holes before came to this solution) but the way it works kind of made more sense to me sinc it doesn't take into account the schema set at all. It's purely about "is this QName the XSD namespace" and "is his a valid XSD primitive local name". I can move it back to SchemaSet if you think it makes sense, but the logic is the same.